### PR TITLE
added indexed events to Delegate + adding ruleOwner

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Quick description:
- [X] Typo/small tweaks

## Changes

- Added indexed events to Delegate (#236)
- Moved struct Rule from Types to Delegate (#230)
- Add ruleOwner to setRule and unsetRule in IDelegate (#210)

## Tests
```
  5 passing (1s)

lerna success run Ran npm script 'test' in 8 packages in 192.5s:
lerna success - @airswap/delegate-factory
lerna success - @airswap/delegate
lerna success - @airswap/index
lerna success - @airswap/indexer
lerna success - @airswap/swap
lerna success - @airswap/types
lerna success - @airswap/wrapper
lerna success - @airswap/order-utils
Done in 200.41s.


## Test Coverage
 Still 100%```

### CODE REVIEW

File: source/delegate/contracts/Delegate.sol [hunk 1]
```diff
@@ -136,13 +136,13 @@ contract Delegate is IDelegate, Ownable {
     * @dev swap needs to be given permission to move funds from the delegate
     * @param senderToken address Token the delgeate will send
     * @param senderToken address Token the delegate will receive
-    * @param rule Types.Rule Rule to set on a delegate
+    * @param rule Rule Rule to set on a delegate
```
Comment (alicevhenshaw) [inline source/delegate/contracts/Delegate.sol]: lol `rule Rule Rule`

File: source/delegate/contracts/Delegate.sol [hunk 2]
```diff
@@ -366,13 +366,14 @@ contract Delegate is IDelegate, Ownable {
...
   ) internal {
-    rules[senderToken][signerToken] = Types.Rule({
+    rules[senderToken][signerToken] = Rule({
       maxSenderAmount: maxSenderAmount,
       priceCoef: priceCoef,
       priceExp: priceExp
     });
 
     emit SetRule(
+      owner(),
```
Comment (alicevhenshaw) [inline source/delegate/contracts/Delegate.sol]: is `owner` or `tradeWallet` more likely to be the thing people want to know?

File: source/delegate/contracts/Delegate.sol (hunk 1 above)
Comment (poiresel) [reply source/delegate/contracts/Delegate.sol]: all the rules =)

File: source/delegate/contracts/Delegate.sol (hunk 2 above)
Comment (poiresel) [reply source/delegate/contracts/Delegate.sol]: good question, my thought was ruleOwner allows for a user / staker to track their delegate whereas tradeWallet is more about fund and could be queries from getting the ruleOwner and querying a delegateFactory.

**Pull request merged**